src/services/pcn-router/src/Ports.cpp

```diff
@@ -159,10 +219,15 @@ void Ports::setIp(const std::string &value) {
     // set new ip
     if (parent_.get_shadow()) {
       int prefix = get_netmask_length(getNetmask());
-      std::function<void()> doThis = [&]{parent_.netlink_instance_router_.set_iface_ip(getName(), value, prefix);};
+      std::function<void()> doThis = [&]{parent_.netlink_instance_router_.add_iface_ip(getName(), value, prefix);};
```

Comment:
Shouldn't it be part of the first commit of this PR where these names are changed in netlink. If you introduce this change here it means the code will not compile after the first commit is applied, what is a pity because it will make git bisect difficult to use.